### ISSUE TITLE
komodo 9 menu hang

### ISSUE THREAD

**Issue by lorenrosen** [Komodo/KomodoEdit#312]
Komodo Edit, version 9.1.0, build 15796, platform linux-x86_64.
(might have happened with 9.0 too)
traversing the menus sometimes hangs on a submenu
to reproduce:
using a mouse--
pull down a menu
move down it until a submenu pops out
menu can't be navigated further until a submenu item is selected or the Esc key is pressed.
sometimes (most often if moving down it quickly) a submenu won't come out. Presumably this is by design. So to reproduce, try moving the mouse slowly.
This is on Ubuntu 14.10

**Comment by cgchoffman** [comment]
As a workaround for now, you can tap **Esc**.  That will bring focus up one level in the menu and make it navigable again.

**Comment by Naatan** [comment]
@cgchoffman's comment made me think of a potential workaround solution - forcing focus when a menu opens. This bug has been around for a long time and I don't really have any idea what's causing it, so a workaround might just be justifyable for now.

**Comment by lorenrosen** [comment]
Seems to me it didn't happen until recently, or at least not so often. Maybe increase the timeout (while a mouse is dragged over the menu item) before a submenu appears?
 


     On Tuesday, June 2, 2015 8:35 AM, Nathan Rijksen <notifications@github.com> wrote:
   

 @cgchoffman's comment made me think of a potential workaround solution - forcing focus when a menu opens. This bug has been around for a long time and I don't really have any idea what's causing it, so a workaround might just be justifyable for now.—

**Comment by cgchoffman** [comment]
This has definitely been around since Komodo 8.  I can find the old bug for it if you're interested.

**Comment by lorenrosen** [comment]
Hmm. My guess is that the timing, and hence the frequency, has changed. Although the timing change could be a result of my Ubuntu or monitor upgrades since I used 8 extensively.
 


     On Tuesday, June 2, 2015 5:43 PM, Carey Hoffman <notifications@github.com> wrote:
   

 This has definitely been around since Komodo 8. I can find the old bug for it if you're interested. —

**Comment by Naatan** [comment]
@mitchell-as assigned to you as you are running Ubuntu and can likely reproduce this issue.

**Comment by mitchell-as** [comment]
Indeed I run into this issue, but it's probably a low-level Mozilla thing and I wouldn't have any idea how to fix that.

**Comment by palmerj** [comment]
Any progress on this issue? It's really annoying!

**Comment by mitchell-as** [comment]
Sorry, no progress :(

**Comment by mitchell-as** [comment]
For some strange reason I cannot reproduce this anymore, either on Komodo 8, 9, or X. (Actually, I'm not sure I've ever seen it happen on X.)

Whoever was experiencing this, does this still happen in Komodo X?

Note, I am running on Ubuntu 14.04.1 with Ubuntu Unity. I've been doing regular software updates and I think I'm fully up to date. I wonder if a Unity or GTK software update silently fixed this problem.

**Comment by Defman21** [comment]
I'm running Ubuntu 16.04, Komodo IDE X and sometimes this happens for me. I'm not able to exit a submenu without pressing Esc.

**Comment by mitchell-as** [comment]
I was able to reproduce it today. I guess there are some days it happens and some days it does not. Weird. Sorry for the noise.

**Comment by Defman21** [comment]
Yeah, probably the Moon affects these menus. I'm able to reproduce it very randomly.

**Comment by mitchell-as** [comment]
@Naatan I'm pretty sure this is a focus issue.

Start Komodo and wait until it's totally done initializing, and then wait some more. Try to access the hamburger menu. No problems.

Quit Komodo and start it up again. This time, try and get to the hamburger menu ASAP, before Komodo completely initializes (fills in places pane, reopens documents, etc.) Now the problem will manifest. Any time you try the hamburger menu again, you get that same problem, even if Komodo finished initializing completely.

**Comment by Naatan** [comment]
If it is focus then simply hiding and showing the window should work around it.
